src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigManager.java

```diff
@@ -112,5 +123,43 @@ public Void call() throws Exception {
         }
 
     }
+    
+    static void reconcileJobsAndBuildConfigs() {
+        logger.info("Reconciling jobs and build configs");
+        List<WorkflowJob> jobs = Jenkins.getActiveInstance().getAllItems(WorkflowJob.class);
+        for (WorkflowJob job : jobs) {
+            BuildConfigProjectProperty property = job.getProperty(BuildConfigProjectProperty.class);
+            if (property != null) {
+                String ns = property.getNamespace();
+                String name = property.getName();
+                if (StringUtils.isNotBlank(ns) && StringUtils.isNotBlank(name)) {
+                    logger.info("Checking job " + job + " runs for BuildConfig " + ns + "/" + name);
+                    OpenShiftClient client = getAuthenticatedOpenShiftClient();
+                    BuildConfig bc = client.buildConfigs().inNamespace(ns).withName(name).get();
+                    if (bc == null) {
+                        try {
+                            ACL.impersonate(ACL.SYSTEM, new NotReallyRoleSensitiveCallable<Void, Exception>() {
+                                @Override
+                                public Void call() throws Exception {
+                                    try {
+                                        deleteInProgress(ns + name);
+                                        job.delete();
+                                    } finally {
+                                        removeJobWithBuildConfigNameNamespace(name, ns);
+                                        Jenkins.getActiveInstance().rebuildDependencyGraphAsync();
+                                        deleteCompleted(ns + name);
+                                    }
+                                    return null;
+                                }
+                            });
+                        } catch (Exception e) {
```

Comment:
is it  enough to just log on Exception?
If yes, can you add a comment in code to explain why?

Reply:
will add comment .... but yes, I believe logging and continuing to next BC vs. throwing exception and leaving method is best